src/main/java/at/hannibal2/skyhanni/data/MaxwellAPI.kt

```diff
@@ -0,0 +1,120 @@
+package at.hannibal2.skyhanni.data
+
+import at.hannibal2.skyhanni.data.jsonobjects.repo.MaxwellPowersJson
+import at.hannibal2.skyhanni.events.ConfigLoadEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.ProfileJoinEvent
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeResets
+import at.hannibal2.skyhanni.utils.StringUtils.trimWhiteSpaceAndResets
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object MaxwellAPI {
+    val config = ProfileStorageData.profileSpecific
+    var currentPower: String? = null
+    var magicalPower = -1
+    var powers = mutableListOf<String>()
+
+    private val group = RepoPattern.group("data.maxwell")
+    private val chatPowerpattern by group.pattern(
+        "chat.power",
+        "§eYou selected the §a(?<power>.*) §e(power )?for your §aAccessory Bag§e!"
+    )
+    private val inventoryPowerPattern by group.pattern(
+        "inventory.power",
+        "§7Selected Power: §a(?<power>.*)"
+    )
+    private val inventoryMPPattern by group.pattern(
+        "inventory.mp",
+        "§7Magical Power: §6(?<mp>[\\d,]+)"
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpaceAndResets().removeResets()
+
+        chatPowerpattern.matchMatcher(message) {
+            val power = group("power")
+            currentPower = byNameOrNull(power) ?: return
+            savePower(currentPower)
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        if (event.inventoryName.contains("Accessory Bag Thaumaturgyr")) {
+            val stacks = event.inventoryItems
+            val selectedPower =
+                stacks.values.find { it.getLore().isNotEmpty() && it.getLore().last() == "§aPower is selected!" }
+                    ?: return
+
+            currentPower = byNameOrNull(selectedPower.getLore().first()) ?: return
+            savePower(currentPower)
+
+        } else if (event.inventoryName.contains("Your Bags")) {
```

Comment:
Make a early return instead of a else if
and make the contains a pattern.
```suggestion

            return
        } 
        if (event.inventoryName.contains("Your Bags")) {
```